src/accounts/add/add-dialog-sender.js

```diff
@@ -0,0 +1,272 @@
+    }
+  ]
+};
+
+function maskApiToken(logzioApiToken) {
+  if (!logzioApiToken) {
+    return '';
+  }
+
+  if (logzioApiToken.length < 6) {
+    return '*'.repeat(logzioApiToken.length);
+  }
+
+  const charsToMask = logzioApiToken.length - 6;
+  return '*'.repeat(charsToMask) + logzioApiToken.substring(charsToMask);
+}
+
+function createSelectableRegionList(apiConfig) {
+  const configuredRegions = apiConfig['regions'];
+  const selectableRegionList = [];
+  for (const region in configuredRegions) {
+    if (!configuredRegions.hasOwnProperty(region)) continue;
+
+    selectableRegionList.push({
+      label: configuredRegions[region]['name'],
+      value: region
+    });
+  }
+
+  return selectableRegionList;
+}
+
+function buildAndSendConfigurationDialog(
+  bot,
+  selectableRegionList,
+  reply,
+  config,
+  callback_id
+) {
+  const accountRegion = config.getLogzioAccountRegion() || 'us-east-1';
+  const apiToken = maskApiToken(config.getLogzioApiToken());
+
+  const dialog = bot
+    .createDialog('Logz.io Configuration', callback_id, 'Save')
+    .addSelect(
+      'Account region',
+      'accountRegion',
+      accountRegion,
+      selectableRegionList
+    )
+    .addText('API Token', 'apiToken', null, {
+      placeholder: apiToken,
+      hint:
+        'Create an API token in your Logz.io Enterprise account or email help@logz.io to request one.'
+    })
+    .addText('Alias', 'alias', null, {
+      placeholder: '',
+      hint:
+        'A Slack alias for your Logz.io account. (Letters, numbers, hyphens, and underscores only.)'
+    });
+
+  bot.replyWithDialog(reply, dialog.asObject(), err => {
+    if (err) {
+      logger.error('Unknown error while replying with dialog', err);
+    }
+  });
+}
+
+function buildAndSendAliasConfigurationDialog(bot, reply, config, callback_id) {
+  const dialog = bot
+    .createDialog('Alias Configuration', callback_id, 'Save')
+    .addText('Alias', 'alias', null, {
+      placeholder: '',
+      hint:
+        'A Slack alias for your Logz.io account. (Letters, numbers, hyphens, and underscores only.)'
+    });
+
+  bot.replyWithDialog(reply, dialog.asObject(), err => {
+    if (err) {
+      logger.error('Unknown error while replying with dialog', err);
+    }
+  });
+}
+
+class AddAccountDialogSender {
+  constructor(teamConfigurationService, apiConfig) {
+    this.teamConfigurationService = teamConfigurationService;
+    this.selectableRegionList = createSelectableRegionList(apiConfig);
+  }
+
+  sendSetupMessage(bot, user, isInitializationPhase) {
+    bot.startPrivateConversation({ user }, (err, convo) => {
+      convo.addMessage(
+        {
+          text: `Okay, I won't add an account now. When you're ready, just type @Alice add account.`
+        },
+        'canceled'
+      );
+
+      convo.addQuestion(
+        messageWithButtons,
+        [
+          {
+            pattern: 'yes',
+            callback: (reply, convo) => {
+              this.teamConfigurationService
```

Comment:
maybe it will be easier to read 
when you have a callback assign it to a function like so:
pattern: 'yes',
callback: addAccountDialogResponse
}

function addAccountDialogResponse(reply ,convo) {

}